### DISCUSSION TITLE
Fields not updating on Options Picker Click

### DISCUSSION THREAD

**Discussion by koneubauer** [Budibase/budibase#11550] [Help] [answered]
I have 5 Textfields inside a Data Provider. The Data Provider uses the values of two Options Pickers in the Data Provider as filter.
When I select a different option, the content of the fields doesn't change. Then, when I select a different option again, the field content changes to that of the before selected option.
For testing purposes, I added a Table to display the content. The table always shows the current and correct content.

As On Click Actions of the Option Pickers i have set Update Field Value of all 5 fields and Refresh Data Provider.

I am using a self-hosted docker-compose with version 2.9.25

**Comment by melohagan** [comment]
Hey @koneubauer 

This sound like you are not using the `{{ Field Value }}` binding to get the "on change" value, rather than the value before the on change:

https://docs.budibase.com/docs/text-inputs#on-change

**Comment by koneubauer** [comment]
Hey,

thanks, thats probably it. However, i dont have the option to select {{ Field Value }}.
These are the available options i can select in the Data Provider.
![image](https://github.com/Budibase/budibase/assets/117980610/d558cec9-9837-483b-8450-651e3541da58)

The Data Provider is inside a Form and the pickers are inside the Data Provider.

**Reply by melohagan** [reply]
<img width="305" alt="Screenshot 2023-08-21 at 11 15 51" src="https://github.com/Budibase/budibase/assets/101575380/33115bb3-7c4f-4764-81d6-d57aaffc8f07">

It'll be at a higher level in the menu, under the options picker, not under form

**Reply by koneubauer** [reply]
![image](https://github.com/Budibase/budibase/assets/117980610/b197c263-d636-4242-a04a-1dd8c18df51d)
It's not there. Is there a way in which the elements have to be arranged?

**Reply by melohagan** [reply]
Is this definitely within the *On change* actions drawer of your options picker?
<img width="301" alt="Screenshot 2023-08-21 at 11 20 12" src="https://github.com/Budibase/budibase/assets/101575380/2a89c609-0b32-49f3-8cbb-8aae6b5a47dd">

**Reply by koneubauer** [reply]
No, sorry. I want to use the updated value of the Options Picker as a filter in the Data Provider. The Options Picker is only for selecting a timeframe. Corresponding to that selected timeframe, textfields inside the Data Provider should show the data.

**Reply by melohagan** [reply]
When a data provider references form field bindings, it will automatically refresh.

I think the issues you may be seeing could be caused by this then:

> As On Click Actions of the Option Pickers i have set Update Field Value of all 5 fields and Refresh Data Provider.

As you are updating field values and refreshing the data provider manually, when it will be automatically doing this itself. Could be causing some unexpected behaviour.

**Reply by koneubauer** [reply]
When I delete the Update Field Value Commands from the On Change Option, the content of the textfields stays the same all the time. The Table which i have showing the data is always correct.

**Reply by melohagan** [reply]
I think a screen recording may help me visualise the problem here! Must be missing something.

**Reply by koneubauer** [reply]
https://github.com/Budibase/budibase/assets/117980610/bd9d12b0-68b3-48b8-8990-c620167e9255

Hey, hope this helps!

**Reply by melohagan** [reply, marked as answer]
Thanks for the recording! When you use *Default values* they only load on screen load. If you want to update field values at a later point, then you will need to use the **Update Form Field** action.

Although I can see the challenge in doing that in this case, because the rows won't have loaded until after the *On change* event.

You could try changing the **Form** type to 'Update', and make sure that your form fields are bound to fields of the form schema (no need for the default values).

<img width="298" alt="Screenshot 2023-08-21 at 12 36 59" src="https://github.com/Budibase/budibase/assets/101575380/321d86ae-3512-475b-b06f-4eff97c92a7a">

<img width="1128" alt="Screenshot 2023-08-21 at 12 37 34" src="https://github.com/Budibase/budibase/assets/101575380/96b1577f-0823-4f2c-bc07-a3a6a0d22f6d">

![change id](https://github.com/Budibase/budibase/assets/101575380/4692f724-d842-4bdd-b839-45eb645221dc)

**It's worth noting that you can't nest forms in new apps going forward**. You can still reproduce this by adding the employee picker with a custom field name to your existing form, and then updating [App state](https://docs.budibase.com/docs/app-state) within the *On change*, and then filtering the Data provider on that app state.

**Reply by koneubauer** [reply]
Thanks a lot!
